sql/sql_sequence.cc

```diff
@@ -769,7 +769,9 @@ longlong SEQUENCE::next_value(TABLE *table, bool second_round, int *error)
     DBUG_RETURN(next_value(table, 1, error));
   }
 
-  if (unlikely((*error= write(table, 0))))
+  if (unlikely(
+          (*error= write(table, global_system_variables.binlog_row_image !=
```

Comment:
Should not it be actually `thd->variables.binlog_row_image` that is checked?

Reply:
Ah yes, I saw other parts of this file used `global_system_variables` to get values due to the lack of a `thd `parameter, but those functions don't have a `table` as a parameter either, but here we do. So using `table->in_use` would make the most sense.